group-coordinator/src/main/java/org/apache/kafka/coordinator/group/streams/TargetAssignmentBuilder.java

```diff
@@ -265,12 +265,15 @@ public TargetAssignmentResult build() throws TaskAssignorException {
         // Compute the assignment.
         GroupAssignment newGroupAssignment;
         if (topology.isReady()) {
+            if (topology.subtopologies().isEmpty()) {
+                throw new IllegalStateException("Subtopologies must be present if topology is ready.");
```

Comment:
Central check. This is implied by `isReady`, but I added the `IllegalStateException` just in case some future updates breaks the internal invariants of `ConfiguredTopology` (and to silence IDE warnings).